src/nspawn/nspawn-bind-user.c

```diff
@@ -91,19 +91,27 @@ static int convert_user(
                 UserRecord *u,
                 GroupRecord *g,
                 uid_t allocate_uid,
+                char *shell,
                 UserRecord **ret_converted_user,
                 GroupRecord **ret_converted_group) {
 
         _cleanup_(group_record_unrefp) GroupRecord *converted_group = NULL;
         _cleanup_(user_record_unrefp) UserRecord *converted_user = NULL;
-        _cleanup_free_ char *h = NULL;
+        _cleanup_free_ char *h = NULL,  *sh = NULL;
         sd_json_variant *p, *hp = NULL, *ssh = NULL;
         int r;
 
         assert(u);
         assert(g);
         assert(user_record_gid(u) == g->gid);
 
+        if (!!shell) {
+                if (streq(shell, "yes"))
+                        sh = u->shell; /* copy the user record shell */
```

Comment:
This is not copied, but `sh` has `_cleanup_free_` attribute.
Please use `const char *` for `sh`, or use `strdup()`.

Reply:
Made `sh` const in b173edc87dd3c01c6cb2ca9bb77092e160377501, thanks.